entity-table/index.js

```diff
@@ -287,7 +392,31 @@ module.exports = {
 								done(err);
 							});
 						};
-					}), callback);
+					}), async (err, results) => {
+						if (err) {
+							callback(err);
+						} else {
+							await new Promise((resolve, reject) => {
+								async.parallelLimit(oldS3Files.map(async (file) => {
+									try {
+										await ls.s3.deleteObject({
+											Bucket: file.bucket,
+											Key: file.key,
+										});
+									} catch (e) {
+										logger.error(e);
+									}
+								}), 10, (err, results) => {
+									if (err) {
+										reject(err);
+									} else {
+										resolve(results);
+									}
+								});
+							});
```

Comment:
I think this needs a `.catch(callback)` in case an error is thrown

Reply:
or wrap it in a try catch.  

if we use `.catch` it probably needs to `.catch(callback).then(results=>callback(null, results))` and the next line removed